src/fft.h

```diff
@@ -28,39 +27,24 @@ namespace FFT
         // This function returns normalized image with swapped quadrants
         PenguinV_Image::Image get() const;
 
-        void resize( uint32_t width_, uint32_t height_ );
-
-        kiss_fft_cpx * data(); // returns a pointer to data
-        const kiss_fft_cpx * data() const;
-        uint32_t width() const; // width of array
-        uint32_t height() const; // height of array
-        bool empty() const; // returns true is array is empty (unullocated)
-
     private:
-        kiss_fft_cpx * _data;
-        uint32_t _width;
-        uint32_t _height;
-
-        void _clean();
-
-        void _copy( const ComplexData & data );
-        void _swap( ComplexData & data );
+        void _allocateData(size_t nBytes) override;
+        void _freeData() override;
+        void _copyData(const BaseComplexData<kiss_fft_cpx> & data) override;
     };
 
     // The class for FFT command execution:
     // - conversion from original domain of data to frequency domain and vice versa
     // - complex multiplication in frequency domain (convolution)
-    class FFTExecutor
+    class FFTExecutor : public BaseFFTExecutor
     {
     public:
         FFTExecutor();
         FFTExecutor( uint32_t width_, uint32_t height_ );
         ~FFTExecutor();
 
-        void initialize( uint32_t width_, uint32_t height_ );
-
-        uint32_t width() const;
-        uint32_t height() const;
+        bool dimensionsMatch( const ComplexData & data) const;
+        using BaseFFTExecutor::dimensionsMatch;
```

Comment:
It saves a little space when doing dimension checking for `FFTExecutor::complexMultiplication()`. Since BaseFFTExecutor isn't a class template, it can't implement `dimensionsMatch( BaseComplexData<dataType>`.

Maybe it would be better to put a global function template in `fft_base.h`? Such as 
```
template<typename dataType>
bool dimensionsMatch(BaseComplexData<dataType>, BaseFFTExecutor)
```
?

Reply:
Sounds great. I'll get on it.

Reply:
Also, I think I will make it a template with two parameters where the second parameter defaults to the first. Something like:
```
template <typename _type1, typname _type2 = _type1>
bool EqualSize( const _type1 & first, const _type2 & second )
{
    return ( first.width() == second.width() ) && ( first.height() == second.height() );
}
```
That way the function can be used to compare `ComplexData` and `FFTExecutor`.

Reply:
Okay. Done.